capsules/src/symmetric_encryption.rs

```diff
@@ -137,9 +137,45 @@ impl<'a, E: SymmetricEncryptionDriver + 'a> Crypto<'a, E> {
             state: Cell::new(CryptoState::IDLE),
         }
     }
+
+    fn aes128_crypt_ctr(&self, state: CryptoState) -> ReturnCode {
+        if self.key_configured.get() && !self.busy.get() && self.state.get() == CryptoState::IDLE {
+            for cntr in self.apps.iter() {
+                cntr.enter(|app, _| {
+                    self.busy.set(true);
+                    self.state.set(state);
+                    app.data_buf.as_ref().map(|slice| {
+                        let len1 = slice.len();
```

Comment:
I got a little lost following the names through this. Maybe `app_data_slice` instead of `slice` and `app_ctr_slice` instead of `slice2` (ditto the lengths)